libs/autofile/autofile.go

```diff
@@ -158,23 +168,22 @@ func (af *AutoFile) openFile() error {
 	return nil
 }
 
+// Size returns the size of the AutoFile. It returns -1 and an error if fails
+// get stats or open file.
+// Opens AutoFile if needed.
 func (af *AutoFile) Size() (int64, error) {
 	af.mtx.Lock()
 	defer af.mtx.Unlock()
 
 	if af.file == nil {
-		err := af.openFile()
-		if err != nil {
-			if err == os.ErrNotExist {
```

Comment:
Why not keep this?

Reply:
see commit msg here  https://github.com/tendermint/tendermint/pull/2764/commits/dca010d8dfed2fa5f533a01361fb935eab410141